src/main/groovy/de/gesellix/gradle/docker/tasks/DockerContainerTask.groovy

```diff
@@ -0,0 +1,332 @@
+    group = "Docker"
+
+    //outputs.upToDateWhen { t -> t.changed == false }
+  }
+
+  @TaskAction
+  def run() {
+    if (!containerName) {
+      throw new GradleException("containerName is mandatory")
+    }
+
+    def config = [:]
+
+    config.Image = image
+    config.HostConfig = [:]
+
+    if (env) {
+      config.Env = env
+    }
+
+    if (cmd) {
+      config.Cmd = cmd
+    }
+
+    if (entrypoint) {
+      config.Entrypoint = entrypoint
+    }
+
+    if (links) {
+      config.HostConfig.Links = links
+    }
+
+    config.HostConfig.Privileged = privileged
+
+    if(ports) {
+      config.ExposedPorts = [:]
+      config.HostConfig.PortBindings = config.HostConfig.PortBindings ?: [:]
+      ports.each { p ->
+        ArrayList<String> parts = p.tokenize(":")
+        String container_port = parts.last().toString()
+        if(container_port.toString().indexOf("/") == -1) {
+          container_port = container_port.toString() + "/tcp"
+        }
+
+        config.ExposedPorts[container_port] =[:]
+
+        if (config.HostConfig.PortBindings[container_port] == null) {
+          config.HostConfig.PortBindings[container_port] = []
+        }
+
+        if(parts.size() == 1) {
+          config.HostConfig.PortBindings[container_port] << [HostIp: "0.0.0.0", HostPort: ""]
+        } else if (parts.size() == 2) {
+          config.HostConfig.PortBindings[container_port] << [HostIp: "0.0.0.0",
+                                                             HostPort: parts.get(0).toString()]
+        } else if (parts.size() == 3) {
+          if (parts.get(1)) {
+            config.HostConfig.PortBindings[container_port] << [HostIp: parts.get(1).toString(),
+                                                               HostPort: parts.get(0).toString()]
+          } else {
+            config.HostConfig.PortBindings[container_port] << [HostIp: "0.0.0.0",
+                                                               HostPort: parts.get(0).toString()]
+          }
+        }
+      }
+    }
+
+    if(volumes) {
+      config.Volumes = [:]
+      config.HostConfig.Binds = []
+      volumes.each { v ->
+        def parts = v.tokenize(":")
+
+        if (parts.size() == 2) {
+          config.Volumes[(String)parts.get(1)] = {}
+          config.HostConfig.Binds << v
+        } else if (parts.size() == 3) {
+          if (!(parts.get(2) in ['ro'])) {
+            throw new IllegalArgumentException('any third argument in bind must be "ro"')
+          }
+          config.Volumes[(String)parts.get(1)] = {}
+          config.HostConfig.Binds << v
+        } else if (parts.size() == 1) {
+          config.Volumes[(String)parts.get(0)] = {}
+        } else {
+          throw new IllegalArgumentException("Malformed volume string: ${v}")
+        }
+      }
+    }
+
+    if (extraHosts) {
+      config.HostConfig.ExtraHosts = extraHosts
+    }
+
+    container = new DockerContainer(getDockerClient(), containerName, image, config)
```

Comment:
Everything before this code is more or less setup for the DockerContainer model, merging the user's config and such. Since it's quite much, I'd prefer it to at least extract it into an own method.